src/Microsoft.PowerShell.Commands.Utility/commands/utility/Get-Error.cs

```diff
@@ -0,0 +1,90 @@
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Management.Automation;
+
+namespace Microsoft.PowerShell.Commands
+{
+    /// <summary>
+    /// Class for Get-Error implementation.
+    /// </summary>
+    [Experimental("Microsoft.PowerShell.Utility.PSGetError", ExperimentAction.Show)]
+    [Cmdlet(VerbsCommon.Get, "Error", HelpUri = "https://docs.microsoft.com/powershell/module/microsoft.powershell.utility/get-error?view=powershell-7&WT.mc_id=ps-gethelp", DefaultParameterSetName = NewestParameterSetName)]
+    public sealed class GetErrorCommand : PSCmdlet
+    {
+        internal const string ErrorParameterSetName = "Error";
+        internal const string NewestParameterSetName = "Newest";
+        internal const string AliasNewest = "Last";
+
+        /// <summary>
+        /// Gets or sets the error object to resolve.
+        /// </summary>
+        [Parameter(Position = 0, ValueFromPipeline = true, ParameterSetName = ErrorParameterSetName)]
+        public PSObject InputObject { get; set; }
+
+        /// <summary>
+        /// Gets or sets the number of error objects to resolve starting with newest first.
+        /// </summary>
+        [Parameter(ParameterSetName = NewestParameterSetName)]
+        [Alias(AliasNewest)]
+        [ValidateRange(1, int.MaxValue)]
+        public int Newest { get; set; } = 1;
+
+        /// <summary>
+        /// Process the error object.
+        /// </summary>
+        protected override void ProcessRecord()
+        {
+            var errorRecords = new List<object>();
+            var index = 0;
+
+            if (InputObject != null)
+            {
+                if (InputObject.BaseObject is Exception || InputObject.BaseObject is ErrorRecord)
+                {
+                    errorRecords.Add(InputObject);
+                }
+            }
+            else
+            {
+                var errorVariable = SessionState.PSVariable.Get("error");
+                var count = Newest;
+                ArrayList errors = (ArrayList)errorVariable.Value;
+                if (count > errors.Count)
+                {
+                    count = errors.Count;
+                }
+
+                while (count > 0)
+                {
+                    errorRecords.Add(errors[index]);
+                    index++;
+                    count--;
+                }
+            }
+
+            index = 0;
+            bool addErrorIdentifier = errorRecords.Count > 1 ? true : false;
+
+            foreach (object errorRecord in errorRecords)
+            {
+                PSObject obj = PSObject.AsPSObject(errorRecord);
+                obj.TypeNames.Insert(0, "System.Management.Automation.ErrorRecord#GetError");
+
+                // Remove some types so they don't get rendered by those formats
+                obj.TypeNames.Remove("System.Management.Automation.ErrorRecord");
```

Comment:
It's not a formatting system issue as it will typically start from index 0 until it finds a formatter.  What I've found through testing, is that if `ErrorRecord` or `Exception` is in the typenames (irrespective that it's not first), the formatting system seems to render that using the out-of-band formatting.

Reply:
Is that something that should be logged as a bug to fix?

Reply:
Agree that this seems unexpected - as a consumer, if the output object is an ErrorRecord, I'd expect to be able to classify it as such by checking `TypeNames`